skimage/transform/_warps.py

```diff
@@ -115,8 +115,9 @@ def resize(image, output_shape, order=None, mode='reflect', cval=0, clip=True,
              "from version 0.19 a ValueError will be raised instead of this "
              "warning.", FutureWarning, stacklevel=2)
 
-    factors = (np.asarray(input_shape, dtype=float) /
-               np.asarray(output_shape, dtype=float))
+    factors = np.divide(input_shape, output_shape)
+    # create copy so input values range stays accessible through image for clip
+    img_in = image
```

Comment:
I see a case where this approach doesn't work: when image dtype is not floating point... In the case of `uint8` for example, values will not be clipped between 0 and 1 as it is supposed to be...
We must call `convert_to_float` here to fix this!

Reply:
Indeed there could still be an issue when `preserve_range` is False :smile: Otherwise values will be clipped between 0 and 255 as expected. I will correct this as you suggested !

Reply:
Also, personally I dislike reassigning values in the middle of a function, so, rather than copy image to image_in then overwriting image, how about just renaming the downstream images? ie below, use `image_smoothed = ndi.gaussian_filter(image, ...)`. What do you think @abouysso?

Reply:
That's in fact an option, but you need to define `image_smoothed` even when `anti_aliasing is False` in this case

Reply:
I instead created `img_bounds` to save input image bounds for later clipping... Though?